src/compas_fab/backends/pybullet/client.py

```diff
@@ -0,0 +1,509 @@
+        self.attached_collision_objects = {}
+        self.disabled_collisions = set()
+
+    def __enter__(self):
+        self.connect()
+        return self
+
+    def __exit__(self, *args):
+        self.disconnect()
+
+    def step_simulation(self):
+        """By default, the physics server will not step the simulation,
+        unless you explicitly send a ``step_simulation``command.  This
+        method will perform all the actions in a single forward dynamics
+        simulation step such as collision detection, constraint solving
+        and integration. The timestep is 1/240 second.
+        """
+        pybullet.stepSimulation(physicsClientId=self.client_id)
+
+    def load_robot(self, urdf_file):
+        """Create a pybullet robot using the input urdf file.
+
+        Parameters
+        ----------
+        urdf_file : :obj:`str` or file object
+            Absolute file path to the urdf file name or file object. The mesh file can be linked by either
+            `"package::"` or relative path.
+
+        Notes
+        -----
+        By default, PyBullet will use the convex hull of any mesh loaded from a URDF for collision detection.
+        Amending the link tag as ``<link concave="yes" name="<name of link>">`` will make the mesh concave
+        for static meshes (see this `example <https://github.com/bulletphysics/bullet3/blob/master/data/samurai.urdf>`_).
+        For non-static concave meshes, replace the OBJ files within the URDF with those generated by ``pybullet.vhacd``.
+        """
+        robot_model = RobotModel.from_urdf_file(urdf_file)
+        robot = Robot(robot_model)
+        robot.client = self
+        with redirect_stdout(enabled=not self.verbose):
+            robot.pybullet_uid = pybullet.loadURDF(urdf_file, useFixedBase=True,
+                                                   physicsClientId=self.client_id,
+                                                   flags=pybullet.URDF_USE_SELF_COLLISION)
+        self._add_ids_to_robot_joints(robot)
+        self._add_ids_to_robot_links(robot)
+        return robot
+
+    def _add_ids_to_robot_joints(self, robot):
+        joint_ids = self._get_joint_ids(robot.pybullet_uid)
+        for joint_id in joint_ids:
+            joint_name = self._get_joint_name(joint_id, robot.pybullet_uid)
+            joint = robot.model.get_joint_by_name(joint_name)
+            pybullet_attr = {'id': joint_id}
+            joint.attr.setdefault('pybullet', {}).update(pybullet_attr)
+
+    def _add_ids_to_robot_links(self, robot):
+        joint_ids = self._get_joint_ids(robot.pybullet_uid)
+        for link_id in joint_ids:
+            link_name = self._get_link_name(link_id, robot.pybullet_uid)
+            link = robot.model.get_link_by_name(link_name)
+            pybullet_attr = {'id': link_id}
+            link.attr.setdefault('pybullet', {}).update(pybullet_attr)
+
+    def remove_configurations_in_collision(self, robot, configurations):
```

Comment:
Minor nitpicking, but I would name this `filter_configurations_in_collision` instead.